core/emulate.c

```diff
@@ -1010,22 +1010,30 @@ em_status_t EMCALL em_decode_insn(struct em_context_t *ctxt, const uint8_t *insn
         ctxt->operand_size = 1;
     }
     if (flags & INSN_GROUP) {
-        opcode_group = &opcode->group[ctxt->modrm.opc];
-        opcode->handler = opcode_group->handler;
-        if (!opcode_group->handler) {
+        const struct em_opcode_t *group_opcode;
```

Comment:
Moving the variable declaration here isn't necessary, plus it's probably declared above for maximum compatibly across compilers

Reply:
C89-compilant compilers and gcc -pendantic would fail on local-scoped variable declarations.

Googling around, most people, like you and me, do seem to prefer local-scoped variable declarations. (Especially when C++ variables are initialized with constructor calls, limiting scope can be advantageous.) However, one remark that did stand out was "[Reusing variable names with a different type is prevented](https://softwareengineering.stackexchange.com/a/119212)". 

Other than that, most compilers will try to limit scope and move variables into registers anyway, so it's a moot point.

Reply:
Thanks, that's interesting. Although I do have some vague memory of C89, I guess C99 was already prevalent when I learned coding. The compilers we use for HAXM (MSVC 2017 and Clang) are new enough to accept C99 syntax, so I wouldn't worry about compatibility with C89. In fact, we may consider enabling C11 support in these compilers some day.

> However, one remark that did stand out was "Reusing variable names with a different type is prevented"

To achieve that, I'd rather avoid writing long functions, so we could easily catch variable naming conflicts in code review ;-)